### PR TITLE
feat: support send-to-claude for `snacks.explorer`

### PR DESCRIPTION
**Pull request by andresthor**
## Add _send-to-claude_ for the `folke/snacks.explorer` to work with the plugin

It looks like `Snacks.explorer` has become the default file browser for LazyVim, so it makes sense we'll start seeing more of those feature requests.

Adds support for [snacks.explorer](https://github.com/folke/snacks.nvim) to claudecode.nvim, bringing feature parity with existing file explorer integrations, such as `nvim-tree`, `neo-tree` and `oil.nvim`.

### Implementation

Follows established patterns in `integrations.lua`:
- Detects `snacks_picker_list` filetype
- Implements `_get_snacks_explorer_selection()` using picker API
- Supports visual mode, toggle selection, and single file selection

### Features

- **Visual mode**: Multi-file selection using V/v/Ctrl-V
- **Security**: Filters root-level files (`/etc/passwd`, `/usr/bin/vim`)
- **Directory handling**: Adds trailing slashes
- **Error handling**: Graceful fallbacks with meaningful messages

### Testing

Comprehensive test coverage:
- Visual mode scenarios
- Edge cases (nil items, empty paths)
- Root protection and directory normalization

### Documentation

- Updated keybinding examples
- Added architecture documentation
- Inline comments for `snacks_picker_list` filetype

Enables snacks.explorer users to add files to Claude's context using familiar vim motions, just like other supported file explorers.

### CODE REVIEW

File: dev-config.lua [hunk 1]
```diff
@@ -24,7 +24,7 @@ return {
       "<leader>as",
       "<cmd>ClaudeCodeTreeAdd<cr>",
       desc = "Add file from tree",
-      ft = { "NvimTree", "neo-tree", "oil" },
+      ft = { "NvimTree", "neo-tree", "oil", "snacks_picker_list" }, -- snacks.explorer uses "snacks_picker_list" filetype
```
Comment (ThomasK33) [inline dev-config.lua:27]: Cool, but this comment doesn't add any context here.

**Review by ThomasK33 — CHANGES_REQUESTED** [review]
Thanks for the PR @andresthor.

Just so you know, I've pushed a fixture for the snacks explorer onto your branch to check if it works.

Can we please remove the 'security' features of omitting specific paths?

**Comment by pdicerbo** [comment]
@ThomasK33 do you know why this has been closed? It would be really useful, `snacks.explorer` is widely used..
